ruby/ql/lib/codeql/ruby/frameworks/ActiveRecord.qll

```diff
@@ -208,11 +213,22 @@ class ActiveRecordSqlExecutionRange extends SqlExecution::Range {
     exists(PotentiallyUnsafeSqlExecutingMethodCall mc |
       this.asExpr().getNode() = mc.getSqlFragmentSinkArgument()
     )
+    or
+    exists(DataFlow::CallNode executeCall |
+      executeCall.getReceiver() = activeRecordConnectionInstance() and
+      executeCall.getMethodName() = "execute" and
+      this = executeCall.getArgument(0) and
+      unsafeSqlExpr(this.asExpr().getExpr())
```

Comment:
```suggestion
    this = activeRecordConnectionInstance().getAMethodCall("execute").getArgument(0) and
    unsafeSqlExpr(this.asExpr().getExpr())
```

We get more benefit from global flow if we stay at the level of API nodes as long as possible before lowering to data-flow nodes.